src/App/Utilities/ThemeManager.cs

```diff
@@ -21,47 +22,31 @@ public static void SetThemeStyle(string name, ResourceDictionary resources)
             {
                 Resources = () => resources;
 
+                var newTheme = NeedsThemeUpdate(name, resources);
+                if (newTheme is null)
+                {
+                    return;
+                }
+
+                var currentTheme = resources.MergedDictionaries.FirstOrDefault(md => md is IThemeResourceDictionary);
+                if (currentTheme != null)
+                {
+                    resources.MergedDictionaries.Remove(currentTheme);
```

Comment:
Worst case scenario we're might be going through the list 3 times (`.Any` + `FirstOrDefault` + `.Remove`), wish we could come up with something faster given how often this will be used.

Reply:
It's not often at all, unless you change themes every little time hehe, before that code you have a check to see if the theme has changed, if not this code is not triggered at all.